lib/preboots/scheduler.js

```diff
@@ -0,0 +1,234 @@
+/* eslint max-params: 0*/
+const EE = require('events');
+const async = require('async');
+const once = require('one-time');
+const semver = require('semver');
+const nsqStream = require('nsq-stream');
+const from = require('from2');
+const util = require('util');
+
+const assign = Object.assign;
+
+util.inherits(Scheduler, EE);
+
+/**
+ * Scheduler
+ * @constructor
+ * @param {Object} options - Configuration object
+ */
+function Scheduler(options) {
+  EE.call(this);
+  this.log = options.log;
+  this.nsq = options.nsq;
+  this.topic = options.topic;
+  this.models = options.models;
+  this.conc = options.concurrency || 5;
+  this.defaultInterval = options.interval || 60 * 60 * 1000;
+
+  this.intervals = new Map();
+}
+
+/**
+ * Set an interval to schedule catch up jobs for the given env
+ * @function setInterval
+ * @param {String} env - Environment
+ * @param {Number} time - milliseconds of interval
+ * @returns {undefined}
+ */
+Scheduler.prototype.setInterval = function interval(env, time) {
+  time = time || this.defaultInterval;
+  const int = setInterval(() => {
```

Comment:
I'd prefer that you name this variable `interval` since `int` could be confused with `integer`.

Reply:
this overwrites the function name in the given scope

Reply:
Ok, then something like `intervalId`